src/combinatorics/derangements/exponential.lean

```diff
@@ -0,0 +1,67 @@
+/-
+Copyright (c) 2021 Henry Swanson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Henry Swanson, Patrick Massot
+-/
+import combinatorics.derangements.finite
+import analysis.complex.basic
+import data.complex.exponential
+import topology.metric_space.cau_seq_filter
+
+/-!
+# Derangement exponential series
+
+This file proves that the probability of a permutation on n elements being a derangement is 1/e.
+The specific lemma is `num_derangements_tendsto_inv_e`.
+-/
+open filter
+open finset
+
+open_locale big_operators
+open_locale topological_space
+
+lemma complex.tendsto_iff_real (u : ℕ → ℝ) (x : ℝ) :
```

Comment:
Yeah, I'm pretty sure they already exist somewhere, but I can't find where :\

Reply:
@hrmacbeth Could you help us out here?

Reply:
These should be added to the library in the more powerful `has_sum` version, not the `tendsto ... (range n)` version.  Here's a fast proof, but I don't know if there's a proof which circumvents @ADedecker's new work on the exponential in Banach spaces (or indeed if they're in the library).  This might be worth asking on Zulip.
```
import analysis.normed_space.exponential

open_locale nat

example (x : ℝ) : has_sum (λ (n : ℕ), x ^ n / n!) (real.exp x) :=
begin
  rw real.exp_eq_exp_ℝ_ℝ,
  exact exp_series_field_has_sum_exp x,
end
```

Then, at the point of use, convert to the form stated by Henry using `has_sum.tendsto_sum_nat`.

Reply:
Your `complex.tendsto_iff_real` is a special case of `complex.isometry_of_real.tendsto_nhds_iff`.

Regarding Heather's comment, I think we indeed still lack this precise lemma about the sum of `x ^ n / n!` for `x : ℝ` and I can't find a short proof without using the general setup.

Reply:
So, I haven't touched much analysis yet (outside of these [tutorials](https://github.com/leanprover-community/tutorials)), so I'm a little out of my depth here. Is this updated proof better?

Reply:
I PRed what was needed for them a while back. Well, actually, everything was already there. I just had to shortcut dot notation a bit: #8693.